sdk/python/kfp/cli/component_test.py

```diff
@@ -443,6 +448,66 @@ def test_docker_file_is_created_correctly(self):
                 COPY . .
                 '''))
 
+    @mock.patch('kfp.__version__', '1.2.3')
+    def test_docker_file_is_created_correctly_with_one_url(self):
+        component = _make_component(
+            func_name='train',
+            target_image='custom-image',
+            pip_index_urls=['https://pypi.org/simple'])
+        _write_components('components.py', component)
+
+        result = self.runner.invoke(
+            self.cli,
+            ['build', str(self._working_dir)],
+        )
+        self.assertEqual(result.exit_code, 0)
+        self._docker_client.api.build.assert_called_once()
+        self.assert_file_exists_and_contains(
+            'Dockerfile',
+            textwrap.dedent('''\
+                # Generated by KFP.
+
+                FROM python:3.7
+
+                WORKDIR /usr/local/src/kfp/components
+                COPY runtime-requirements.txt runtime-requirements.txt
+                RUN pip install --index-url https://pypi.org/simple --trusted-host https://pypi.org/simple --no-cache-dir -r runtime-requirements.txt
+
+                RUN pip install --index-url https://pypi.org/simple --trusted-host https://pypi.org/simple --no-cache-dir kfp==1.2.3
+                COPY . .
+                '''))
+
+    @mock.patch('kfp.__version__', '1.2.3')
+    def test_docker_file_is_created_correctly_with_two_urls(self):
+        component = _make_component(
+            func_name='train',
+            target_image='custom-image',
+            pip_index_urls=[
+                'https://pypi.org/simple', 'https://example.com/pypi/simple'
+            ])
+        _write_components('components.py', component)
+
+        result = self.runner.invoke(
+            self.cli,
+            ['build', str(self._working_dir)],
+        )
+        self.assertEqual(result.exit_code, 0)
+        self._docker_client.api.build.assert_called_once()
+        self.assert_file_exists_and_contains(
+            'Dockerfile',
+            textwrap.dedent('''\
+                # Generated by KFP.
+
+                FROM python:3.7
+
+                WORKDIR /usr/local/src/kfp/components
+                COPY runtime-requirements.txt runtime-requirements.txt
+                RUN pip install --index-url https://pypi.org/simple --trusted-host https://pypi.org/simple  --extra-index-url https://example.com/pypi/simple --trusted-host https://example.com/pypi/simple --no-cache-dir -r runtime-requirements.txt
```

Comment:
(not your code; feel free to ignore) Looks like string joiner implementation results in two spaces between the first entry and the second (before the first `--extra-index-url`).

Reply:
Fixed